lib/auth_page.dart

```diff
@@ -66,6 +83,56 @@ Stream<int> _signin(BuildContext context) async* {
   }
 }
 
+void _facebookSignout() {
+  try {
+    final _facebookLogin = FacebookLogin();
+    _facebookLogin.logOut();
+  } catch (error, trace) {
+    // reset the UI and display an alert
+  }
+}
+
+Stream<int> _facebookSignin(BuildContext context) async* {
+  try {
+    final _fireAuth = FirebaseAuth.instance;
+    final facebookLogin = FacebookLogin();
+    final result = await facebookLogin.logIn(['email']);
+
+    switch (result.status) {
+      case FacebookLoginStatus.loggedIn:
+        /// the auth info will be picked up by the listener on [onAuthStateChanged]
+        /// and emitted by [streamOfStateChanges]
+        
+        // signal to change UI
+        yield 2;
+
+        final credential = FacebookAuthProvider.getCredential(accessToken: result.accessToken.token);
+        await _fireAuth.signInWithCredential(credential);
+
+        // we are signed in so reset the UI
+        yield 0;
+        break;
+      case FacebookLoginStatus.cancelledByUser:
+        // _showCancelledMessage();
+        yield 0;
+        break;
+      case FacebookLoginStatus.error:
+        // _showErrorOnUI(result.errorMessage);
+        yield 0;
+        break;
+    }
+
+  } catch (error, trace) {
+    // reset the UI and display an alert
+
+    yield 0;
+    // errors with code kSignInCanceledError are swallowed by the
+    // GoogleSignIn.signIn() method so we can assume anything caught here
+    // is unexpected and for display
+    _showDialog(context);
```

Comment:
Can we display the error message in the dialog here?

Reply:
I'm not going to disagree though I don't think I want to make up the scaffolding of better error handling with this branch. This can be a later issue once the building blocks of error handling is sorted out. My amature dart I'm not willing to pollute this PR with my ugly attempt of this. 

If I'm mistaken and there is some good examples in the codebase I'm happy to follow the patterns and add this.

Reply:
The error is available at the point the _showDialog function is called so all that would be required is to make the _showDialog function take a string parameter and display it.  Again, I could easily be missing something here...